### ISSUE TITLE
Question regarding gradient accumulation

### ISSUE THREAD

**Issue by yuanzhi-zhu** [valeoai/Maskgit-pytorch#22]
Thank you so much for your contribution to this community.
Recently I just spot a possible error of the implementation of the gradient accumulation.
https://github.com/valeoai/Maskgit-pytorch/blob/b0b2b3cc11cffd0b159f22dc1c6e73a7e8b53db3/Trainer/vit.py#L197-L202
In this piece of code, it seems that every time we update the model parameters, we will throwing away the previous accumulated gradients before using them.

**Comment by llvictorll** [comment]
Hello, 

Yes, I found it out a couple of months ago, I will also correct it in the near future. And also add new pre-trained networks.

Thanks, 

Victor
